Mathlib/Topology/ContinuousFunction/Bounded.lean

```diff
@@ -789,6 +789,74 @@ theorem coe_sub : ⇑(f - g) = f - g := rfl
 
 end sub
 
+section casts
+
+variable [TopologicalSpace α] {β : Type*} [PseudoMetricSpace β]
+
+instance [NatCast β] : NatCast (α →ᵇ β) := ⟨fun n ↦ BoundedContinuousFunction.const _ n⟩
+
+@[simp]
+theorem natCast_apply [NatCast β] (n : ℕ) (x : α) : (n : α →ᵇ β) x = n := rfl
+
+instance [IntCast β] : IntCast (α →ᵇ β) := ⟨fun m ↦ BoundedContinuousFunction.const _ m⟩
+
+@[simp]
+theorem intCast_apply [IntCast β] (m : ℤ) (x : α) : (m : α →ᵇ β) x = m := rfl
+
+end casts
+
+section mul
+
+variable [TopologicalSpace α] {R : Type*} [PseudoMetricSpace R]
+
+instance instMul [Mul R] [BoundedMul R] [ContinuousMul R] :
+    Mul (α →ᵇ R) where
+  mul f g := {
+    toFun := fun x ↦ f x * g x
+    continuous_toFun := f.continuous.mul g.continuous
+    map_bounded' := mul_bounded_of_bounded_of_bounded (map_bounded f) (map_bounded g) }
+
+@[simp]
+theorem coe_mul [Mul R] [BoundedMul R] [ContinuousMul R] (f g : α →ᵇ R) : ⇑(f * g) = f * g := rfl
+#align bounded_continuous_function.coe_mul BoundedContinuousFunction.coe_mul
+
+theorem mul_apply [Mul R] [BoundedMul R] [ContinuousMul R] (f g : α →ᵇ R) (x : α) :
+    (f * g) x = f x * g x := rfl
+#align bounded_continuous_function.mul_apply BoundedContinuousFunction.mul_apply
+
+instance instPow [Monoid R] [BoundedMul R] [ContinuousMul R] : Pow (α →ᵇ R) ℕ where
+  pow f n := {
+    toFun := fun x ↦ (f x) ^ n
+    continuous_toFun := f.continuous.pow n
+    map_bounded' := by
+      obtain ⟨C, hC⟩ := Metric.isBounded_iff.mp <| isBounded_pow (isBounded_range f) n
+      exact ⟨C, fun x y ↦ @hC ((f x)^n) (by simp) ((f y)^n) (by simp)⟩ }
+
+theorem coe_pow [Monoid R] [BoundedMul R] [ContinuousMul R] (n : ℕ) (f : α →ᵇ R) :
+    ⇑(f ^ n) = (⇑f) ^ n := rfl
+#align bounded_continuous_function.coe_pow BoundedContinuousFunction.coe_pow
+
+@[simp]
+theorem pow_apply [Monoid R] [BoundedMul R] [ContinuousMul R] (n : ℕ) (f : α →ᵇ R) (x : α) :
+    (f ^ n) x = f x ^ n := rfl
+#align bounded_continuous_function.pow_apply BoundedContinuousFunction.pow_apply
+
+instance instMonoid [Monoid R] [BoundedMul R] [ContinuousMul R] :
+    Monoid (α →ᵇ R) :=
+  Injective.monoid (↑) DFunLike.coe_injective' rfl (fun _ _ ↦ rfl) (fun _ _ ↦ rfl)
+
+instance instCommMonoid [CommMonoid R] [BoundedMul R] [ContinuousMul R] :
+    CommMonoid (α →ᵇ R) where
+  __ := instMonoid
+  mul_comm f g := by ext x; simp [mul_apply, mul_comm]
+
+instance instSemiring [Semiring R] [BoundedMul R] [ContinuousMul R] [LipschitzAdd R] :
```

Comment:
I agree.

However, the addition operation feels less urgent than the multiplication and subtraction, since at least I am not aware of a case where it doesn't work (whereas the lack of subtraction and multiplication have been holding back Riesz-Markov-Kakutani contributions). So this seems like an issue of consistency, which probably does not affect any uses at the moment.

I will add a TODO and make a PR about that consistent design soon, deriving an instance of `BoundedAdd` from `LipschitzAdd`, as that should already cover all existing and planned use cases that I am aware of. I also just realized that I did not use Lipschitzness of subtraction in the BoundedSub instances as I had intended! So I will also include simplifications to instances of `BoundedSub` at the same time.

Reply:
Just for the record, I am going to address the consistency of the addition instances with the other operations on bounded continuous functions --- I will return to the WIP PR #13035 in the near future. Thank you again for this improvement suggestion!